### PR TITLE
Add `EnvironmentBuilder` and remove `Manager`

### PR DESCRIPTION
**Pull request by kinrezC**
**Give an overview of the tasks completed**
- Remove `manager.rs`
- Add `EnvironmentBuilder` struct for easy `Environment` and `EnvironmentParameter` configuration
- `Environment` `label: String` -> `label: Option<String>`
- Refactor tests to remove `manager` 

**Link to issue(s) that this PR closes**
For #513

### CODE REVIEW

**Review by 0xJepsen — APPROVED** [review]
@kinrezC I must say, and i don't say this often, but i think think this is the right abstraction. Really awesome work on this

File: arbiter-core/src/environment.rs [hunk 1]
```diff
@@ -287,6 +329,58 @@ pub enum EnvironmentError {
...
+        self.gas_settings = gas_settings;
+        self
+    }
+
+    /// Sets the `label` for the `EnvironmentBuilder`.
+    /// This is an optional string that can be used to identify the [`Environment`].
+    pub fn label(mut self, label: String) -> Self {
+        self.label = Some(label);
+        self
+    }
```
Comment (Autoparallel) [inline arbiter-core/src/environment.rs:364]: Can we do this instead?
```suggestion
    pub fn label(mut self, label: impl Into<String>) -> Self {
        self.label = Some(label.into());
        self
    }
```

File: arbiter-core/src/environment.rs [hunk 2]
```diff
@@ -698,6 +792,77 @@ impl Environment {
...
+    /// * `Err(EnvironmentError::Pause(String))` if the environment is stopped or in an invalid state.
+    pub fn pause(&mut self) -> Result<(), EnvironmentError> {
+        match self.state.load(std::sync::atomic::Ordering::SeqCst) {
+            State::Running => {
+                self.state
+                    .store(State::Paused, std::sync::atomic::Ordering::SeqCst);
+                info!(
+                    "Pausing environment with label: {}",
+                    self.parameters.label.as_ref().unwrap_or(&"".into())
+                );
```
Comment (Autoparallel) [inline arbiter-core/src/environment.rs:815]: Can we do:
```suggestion
                if let Some(label) {
                info!(
                    "Pausing environment with label: {}",
                    label
                );
                } else {
                info!("Pausing environment...");
                }
```

File: arbiter-core/src/environment.rs [hunk 3]
```diff
@@ -698,6 +792,77 @@ impl Environment {
...
+    /// * `Err(EnvironmentError::Stop(String))` if the environment is in an invalid state.
+    pub fn stop(&mut self) -> Result<(), EnvironmentError> {
+        match self.state.load(std::sync::atomic::Ordering::SeqCst) {
+            State::Running => {
+                self.state
+                    .store(State::Stopped, std::sync::atomic::Ordering::SeqCst);
+                info!(
+                    "Stopping environment with label: {}",
+                    self.parameters.label.as_ref().unwrap_or(&"".into())
+                );
```
Comment (Autoparallel) [inline arbiter-core/src/environment.rs:847]: Can we do:
```suggestion
                                if let Some(label) {
                info!(
                    "Pausing environment with label: {}",
                    label
                );
                } else {
                info!("Pausing environment...");
                }
```

File: arbiter-core/src/environment.rs [hunk 4]
```diff
@@ -913,9 +1078,10 @@ pub(crate) struct Socket {
 /// their own external API and the latter will allow the end user to set
 /// a rate parameter and seed for a Poisson distribution that will be
 /// used to sample the amount of transactions per block.
-#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Default)]
 pub enum BlockSettings {
     /// The block number will be controlled by the end user.
+    #[default]
```
Comment (Autoparallel) [inline arbiter-core/src/environment.rs:1084]: This is a new trick for me!

File: arbiter-core/src/environment.rs [hunk 5]
```diff
@@ -1020,15 +1187,37 @@ pub(crate) mod tests {
...
+    fn new_with_builder() {
+        let environment = EnvironmentBuilder::new().build();
+        assert_eq!(environment.parameters.label, None);
+        let state = environment.state.load(std::sync::atomic::Ordering::SeqCst);
+        assert_eq!(state, State::Initialization);
+    }
+    #[test]
+    fn new_with_builder_custom_settings() {
+        let environment = EnvironmentBuilder::new()
+            .label(TEST_ENV_LABEL.into())
```
Comment (Autoparallel) [inline arbiter-core/src/environment.rs:1200]: with other change you can drop `.into()` here.

Comment (Autoparallel) [reply arbiter-core/src/environment.rs:1200]: And everywhere else.

**Review by Autoparallel — CHANGES_REQUESTED** [review]
This looks super good! I love it.

I left some comments and suggestions but nothing major.

**Review by Alexangelj — COMMENTED** [review]
I only did a high level overview of the changes - overall this is a much better dev experience, after experiencing how to figure out how to get all the default values I need to initialize the environment, it's nice its all handled for me

**Pull request merged**